app/assets/stylesheets/header.scss

```diff
@@ -145,4 +116,97 @@
       margin-right: 10px;
     }
   }
+
+  @media (max-width: $grid-float-breakpoint-max) {
+
+    .navbar-nav.hidden-xs {
+      margin: 0;
+      margin-top: 6px;
+    }
+
+    .nav-badge {
+      color: $navbar-inverse-link-color;
+      &:hover { color: $navbar-inverse-link-hover-color; }
+      &:hover,
+      &:focus {
+        background-color: transparent;
+      }
+    }
+
+    #navbar-collapse {
+      .form-group,
+      .twitter-typeahead {
+        display: block !important;
+        margin-bottom: 0;
+
+        &,
+        input { width: 100%; }
+      }
+    }
+
+    .nav-badges .dropdown-menu {
+      width: 300px;
+    }
+
+    #user_menu.dropdown-menu {
+      display: block;
+      position: static;
+      width: 100%;
+      background-color: transparent;
+      box-shadow: none;
+      border: 0;
+      padding: 0;
+      margin-top: -8px; // To compensate parent ul margin
+
+      a {
+        color: #9d9d9d;
```

Comment:
Color literals like `#9d9d9d` should only be used in variable declarations; they should be referred to via variable everywhere else.